src/Underscores.jl

```diff
@@ -72,18 +72,27 @@ function lower_underscores(ex)
     if ex isa Expr
         if isquoted(ex)
             return ex
+
         elseif ex.head == :call && length(ex.args) > 1 &&
                ex.args[1] in _pipeline_ops
             # Special case for pipelining and composition operators
             return Expr(ex.head, ex.args[1],
                         map(lower_underscores, ex.args[2:end])...)
+
         elseif ex.head == :.       && length(ex.args) == 2 &&
                ex.args[2] isa Expr && ex.args[2].head == :tuple
             # Broadcast calls treated as normal calls for underscore lowering
             return replace__(Expr(ex.head, replace_(ex.args[1]),
                                   Expr(:tuple, map(replace_, ex.args[2].args)...)))
+
+        elseif ex.head == :ref && ex.args[1] isa Expr
+            # Indexing is not counted as outermost function
+            return replace__(Expr(ex.head,
+                Expr(ex.args[1].head, map(replace_, ex.args[1].args)...), ex.args[2]))
+
         elseif ex.head == :do
             error("@_ expansion for `do` syntax is reserved")
+
```

Comment:
Minor nit: I'd somewhat prefer it without these extra blank lines.

Reply:
Done, sorry.